File: java/ql/src/experimental/Security/CWE/CWE-1427/SystemPromptInjectionGood.java
import javax.servlet.http.HttpServletRequest;
import dev.langchain4j.data.message.SystemMessage;
import dev.langchain4j.data.message.UserMessage;

public class SystemPromptInjectionGood {
  private HttpServletRequest request;

  public void build() {
    String userInput = request.getParameter("persona");
    // GOOD: the system prompt is trusted, and untrusted input is confined to a user message.
    SystemMessage system = SystemMessage.from("You are a helpful assistant.");
    UserMessage user = UserMessage.from(userInput);
  }
}

File: csharp/ql/src/experimental/CWE-1427/SystemPromptInjectionGood.cs
using System.Web.Mvc;
using OpenAI.Chat;

public class GoodController : Controller
{
    [HttpPost]
    public ActionResult Configure(string persona)
    {
        // GOOD: the system prompt is trusted, and untrusted input is confined to a user message.
        var system = new SystemChatMessage("You are a helpful assistant.");
        var user = new UserChatMessage(persona);
        return View();
    }
}
